Index: app/src/main/java/com/example/esp32server/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/esp32server/MainActivity.java b/app/src/main/java/com/example/esp32server/MainActivity.java
--- a/app/src/main/java/com/example/esp32server/MainActivity.java	
+++ b/app/src/main/java/com/example/esp32server/MainActivity.java	(date 1720881263831)
@@ -5,6 +5,7 @@
 import android.location.Location;
 import android.os.AsyncTask;
 import android.os.Bundle;
+import android.os.Handler;
 import android.widget.Toast;
 
 import androidx.annotation.NonNull;
@@ -23,8 +24,11 @@
 
 public class MainActivity extends AppCompatActivity {
 
+
     private static final int LOCATION_PERMISSION_REQUEST_CODE = 1;
     private FusedLocationProviderClient fusedLocationClient;
+    private Handler handler;
+    private Runnable runnable;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
@@ -35,6 +39,26 @@
 
         // Verifica e solicita permissão de localização
         checkLocationPermission();
+
+        // Inicia o envio da localização a cada 1 segundo
+        handler = new Handler();
+        runnable = new Runnable() {
+            @Override
+            public void run() {
+                getLastKnownLocation();
+                handler.postDelayed(this, 1000); // 1000 milissegundos = 1 segundo
+            }
+        };
+        handler.postDelayed(runnable, 1000); // Inicia o runnable imediatamente
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        // Remove o callback do handler ao destruir a atividade
+        if (handler != null && runnable != null) {
+            handler.removeCallbacks(runnable);
+        }
     }
 
     private void checkLocationPermission() {
Index: app/src/main/java/com/example/esp32server/LocationService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/esp32server/LocationService.java b/app/src/main/java/com/example/esp32server/LocationService.java
new file mode 100644
--- /dev/null	(date 1720881526033)
+++ b/app/src/main/java/com/example/esp32server/LocationService.java	(date 1720881526033)
@@ -0,0 +1,4 @@
+package com.example.esp32server;
+
+public class LocationService {
+}
